src/include_graph_dependencies.cpp

```diff
@@ -5,6 +5,10 @@
 namespace clangmetatool {
 
 namespace {
+  
+// Backup copy of usage_reference_count, only used for internal implementation
+std::map<clangmetatool::types::FileGraphEdge, size_t> usage_reference_count_bk;
```

Comment:
Ah, I meant this map shouldn't be a global, but something that the client holds onto and makes available when a restore is needed. Is that even necessary though, or can one just use `IncludeGraphData::operator=` and keep around an initial object that gets copied back later?

Reply:
Eg:

```
IncludeGraphData copy = collector->getData();
// Use copy to do operations

// Reset copy
copy = collector->getData()
// Use copy to do operations
```

Reply:
That makes sense, I may let `backup` to return a copy and `restore` to consume that copy